GNUmakefile

```diff
@@ -79,3 +79,44 @@ install: build ## install library and headers
 	install .build/libuplink.a ${DESTDIR}/lib
 	install -m 644 .build/uplink/* ${DESTDIR}/include/uplink
 	install -m 644 .build/libuplink.pc ${DESTDIR}/lib/pkgconfig
+
+
+OS ?= $(shell uname)
```

Comment:
I wonder whether we could use a docker image to do this?

For darwin builds when it's done on mac, then we could mount the local libraries and stuff.

Reply:
No.  Cross Compilation would work except Go explicitly prohibits it for legal reasons when there's linking.  Similarly you can't run OSX VMs.  The End User License Agreement (EULA) for Apple Mac OS X legally and explicitly binds the installation and running of the operating system to Apple-labeled computers only."  https://kb.vmware.com/s/article/1000131.  We could have a cloud service do the builds on real Apple hardware. Because this is not on a CI path and is done infrequently, this seems to be an okay 80% solution.

Reply:
I meant that when running on Mac OS you mount the local SDK into a linux VM, which then uses it to build the shared library. Although, I'm not sure, that might be disallowed by the SDK as well. So you would still be using Mac hardware and SDK intended for that machine, however inside the local VM.

Reply:
As I recall, we've gone through this before... https://github.com/storj/docker-golang/blob/2e1eb2d53a37590c1d55791bc1ef54922cd37bb4/Dockerfile#L47

Reply:
That's a different approach that packages up Mac OSX such that it can run on non-mac hardware / VM.

Reply:
Either way, we can find a better solution later.